### PR TITLE
 Setup CircleCI builds

### PR DESCRIPTION
**Pull request by rusmonster**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+    resource_class: xlarge
+    environment:
+      JVM_OPTS: -Xmx15g
+
+commands:
+  download_dependencies:
+    description: Download Dependencies
+    steps:
+      - restore_cache:
+          key: jars-{{ checksum "build.gradle" }}-{{ checksum  "app/build.gradle" }}
```
Comment (berkus) [inline .circleci/config.yml]: Why two spaces after `checksum`

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+      - run:
+          name: Download Dependencies
+          command: ./gradlew androidDependencies
+      - save_cache:
+          paths:
+            - ~/.gradle
+          key: jars-{{ checksum "build.gradle" }}-{{ checksum  "app/build.gradle" }}
```
Comment (berkus) [inline .circleci/config.yml]: Here too.

File: .circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+
+  generate_gcloud_settings:
+    description: "Generate GCloud settings"
+    steps:
+      - run:
+          name: Decode Google Cloud key.json
+          command: echo "$GCLOUD_SETTINGS" | base64 --decode > rtd-shared-lib-a27041a3fda9.json
```
Comment (berkus) [inline .circleci/config.yml]: Hmm rtd-shared-lib?

File: .circleci/config.yml [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+      - run:
+          name: Set Google Cloud target project
+          command: gcloud config set project rtd-shared-lib
+      - run:
+          name: Authenticate with Google Cloud
+          command: gcloud auth activate-service-account  rtd-shared-lib@appspot.gserviceaccount.com --key-file=rtd-shared-lib-a27041a3fda9.json --project=rtd-shared-lib
```
Comment (berkus) [inline .circleci/config.yml]: Move acc name to env var? Generate json file name from same var.

**Review by berkus — CHANGES_REQUESTED** [review]

File: .circleci/config.yml (hunk 1 above)
Comment (rusmonster) [reply .circleci/config.yml]: fixed

File: .circleci/config.yml (hunk 2 above)
Comment (rusmonster) [reply .circleci/config.yml]: too

File: .circleci/config.yml (hunk 3 above)
Comment (rusmonster) [reply .circleci/config.yml]: fixed

File: .circleci/config.yml (hunk 4 above)
Comment (rusmonster) [reply .circleci/config.yml]: json file name is hardcoded, project name moved to env var

**Review by berkus — APPROVED** [review]

**Pull request merged**